### PR TITLE
dcp-386 search by keywords

### PR DESCRIPTION
**Pull request by amnonkhen**
ebi-ait/dcp-ingest-central#386

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,20 @@ You will need a mongo database running on localhost with default ports to run th
 ## Running locally
 You can run ingest-core on your local system with the script
 
+### populating the local DB
+
+You could populate the local DB with prod data using the following script:
```
Comment (aaclan-ebi) [inline README.md:31]: thanks for including this in the PR!

File: src/main/java/org/humancellatlas/ingest/project/ProjectQueryBuilder.java [hunk 2]
```diff
@@ -28,10 +31,37 @@ public static Query buildProjectsQuery(SearchFilter searchFilter) {
...
         Optional.ofNullable(searchFilter.getSearch())
+                .map(search -> ProjectQueryBuilder.formatSearchString(searchFilter))
                 .ifPresent(search -> {
                     query.addCriteria(TextCriteria.forDefaultLanguage().matching(String.valueOf(search)));
                 });
 
         return query;
     }
+
+    private static final Map<SearchType, Function<SearchFilter, String>> x = Map.of(
```
Comment (aaclan-ebi) [inline src/main/java/org/humancellatlas/ingest/project/ProjectQueryBuilder.java:42]: could rename `x` to be `formatter`  for readability?

**Review by aaclan-ebi — APPROVED** [review]

File: src/test/java/org/humancellatlas/ingest/project/ProjectFilterQueryBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+public class ProjectFilterQueryBuilderTest {
+    @Test
+    void null_search_type_with_non_null_text() {
+        SearchFilter searchFilter = new SearchFilter("project keyword",null,null,null);
+        ProjectQueryBuilder.buildProjectsQuery(searchFilter);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
```
Comment (jacobwindsor) [inline src/test/java/org/humancellatlas/ingest/project/ProjectFilterQueryBuilderTest.java:20]: This is coool

**Review by jacobwindsor — APPROVED** [review]

File: src/main/java/org/humancellatlas/ingest/project/ProjectQueryBuilder.java (hunk 2 above)
Comment (amnonkhen) [reply src/main/java/org/humancellatlas/ingest/project/ProjectQueryBuilder.java:42]: I wanted to find a good name and forgot :(

Comment (aaclan-ebi) [reply src/main/java/org/humancellatlas/ingest/project/ProjectQueryBuilder.java]: i think you picked good names for it!

**Review by aaclan-ebi — APPROVED** [review]

**Review by jacobwindsor — APPROVED** [review]

File: src/integration/java/org/humancellatlas/ingest/project/web/ProjectControllerTest.java [hunk 4]
```diff
@@ -146,4 +155,83 @@ void onlyUpdateAllowedFields() throws Exception {
...
+        public void allValuesSetSomeNull(String search, String wrangler, String wranglingState, String searchType) throws Exception {
+            //given:
+            var content = Map.of(
+                    "search", search,
+                    "wrangler", wrangler,
+                    "wranglingState", wranglingState,
+                    "searchType", searchType
+            );
+
+            webApp
```
Comment (ke4) [inline src/integration/java/org/humancellatlas/ingest/project/web/ProjectControllerTest.java:192]: This is very similar to a MockMVC test in Spring 👍

**Review by ke4 — APPROVED** [review]
LGTM 👍

**Review by aaclan-ebi — APPROVED** [review]

**Pull request merged**